ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -1347,13 +1347,28 @@ module HostsConfFunc (T : LocalHostTag) : HostsConf = struct
     let name = String.lowercase_ascii name in
     let domain = String.lowercase_ascii domain in
     let fqdn = Printf.sprintf "%s.%s" name domain in
+    let rec add_hostname pre line =
+      match line with
+      | ip :: alias when ip = T.local_ip ->
+          (* Add localhost IP *)
+          add_hostname [ip] alias
+      | sp :: left when sp = "" ->
+          (* Add space to reserve the indent *)
+          add_hostname (pre @ [sp]) left
+      | alias :: left ->
+          (* hosts entry: ip fqdn alias1 alias2 ... *)
+          pre @ [fqdn; name; alias] @ left
+      | _ ->
```

Comment:
This should be the empty list, it's better to be explicit when enumerating pattern matches, cases tend to be missed when modifying existing code
```suggestion
      | [] ->
```